compat/mingw.c

```diff
@@ -312,7 +313,10 @@ int mingw_core_config(const char *var, const char *value, void *cb)
 		core_fscache = git_config_bool(var, value);
 		return 0;
 	}
-
+	if (!strcmp(var, "core.wslcompat")) {
+		wsl_compat = git_config_bool(var, value);
+		return 0;
+	}
```

Comment:
Hmm. I think that parsing the config value in `mingw_core_config()` will run into the same problems as we had with the long paths support. The reason is that it is too easy to miss a code path where `git_default_config()` is _not_ called.

Therefore, I really would like to see [`are_long_paths_enabled()`](https://github.com/git-for-windows/git/blob/v2.41.0.windows.1/compat/mingw.c#L280-L299) imitated instead.